http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -336,22 +343,33 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
     } yield ()
 
   private[monitor] def handleCreateDiskMessage(msg: CreateDiskMessage, now: Instant)(
-    implicit traceId: ApplicativeAsk[F, AppContext]
+    implicit ev: ApplicativeAsk[F, AppContext]
   ): F[Unit] = {
     val createDisk = for {
-      diskResult <- googleDiskService.createDisk(
+      ctx <- ev.ask
+      operation <- googleDiskService.createDisk(
         msg.googleProject,
         msg.zone,
         Disk
           .newBuilder()
           .setName(msg.name.value)
-          .setSizeGb(msg.size.asString)
+          .setSizeGb(msg.size.gb.toString)
           .setZone(msg.zone.value)
-          .setType(msg.diskType.googleString)
+          .setType(msg.diskType.googleString(msg.googleProject, msg.zone))
           .setPhysicalBlockSizeBytes(msg.blockSize.bytes.toString)
           .build()
       )
-      _ <- persistentDiskQuery.updateGoogleId(msg.diskId, GoogleId(diskResult.getTargetId), now).transaction[F]
+      _ <- persistentDiskQuery.updateGoogleId(msg.diskId, GoogleId(operation.getTargetId), now).transaction[F]
+      task = for {
+        _ <- streamFUntilDone(F.pure(operation), 5, 5 seconds).compile.drain
```

Comment:
how come this delay is 5 and the other two are 10 seconds?

Reply:
this will be updated in the follow up PR....I don't have a good sense how long the interval should be right now...5 and 10 seconds is just random guess...